ibis/tests/expr/test_value_exprs.py

```diff
@@ -626,20 +627,23 @@ def test_null_column_union():
     assert s.union(t.mutate(b=ibis.NA.cast('double'))).schema() == s.schema()
 
 
-def test_string_compare_numeric_array(table):
-    with pytest.raises(TypeError):
-        table.g == table.f  # noqa: B015
+# TODO(kszucs): rethink how to handle this, theoretically the user should be
+# able to compare a string column with a numeric literal which would evaluate
+# to False for all rows
```

Comment:
I do not agree.

Why wouldn't it be a type error? Why would `'a' == 97` be better defined than say `'a' + 97`?

Reply:
Restoring the original behaviour. Removed the comment.